csharp/extractor/Semmle.Extraction/Entities/SourceLocation.cs

```diff
@@ -1,58 +1,64 @@
+using System;
 using System.IO;
 using Microsoft.CodeAnalysis;
 
 namespace Semmle.Extraction.Entities
 {
-    public class SourceLocation : Location
+    public abstract class SourceLocation : Location {
```

Comment:
Why this abstract class? It appears to only have `NonGeneratedSourceLocation` as a sub class.

Reply:
`GeneratedLocation` also extends `SourceLocation`, and it's `abstract` for the `WriteId` and `Populate` methods.